src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/Delegation/DelegatedCompletionListProvider.cs

```diff
@@ -112,6 +118,24 @@ internal class DelegatedCompletionListProvider
         return rewrittenResponse;
     }
 
+    private async Task<bool> ShouldIncludeSnippetsAsync(VersionedDocumentContext documentContext, int absoluteIndex, CancellationToken cancellationToken)
+    {
+        var codeDocument = await documentContext.GetCodeDocumentAsync(cancellationToken).ConfigureAwait(false);
+        var tree = codeDocument.GetSyntaxTree();
+
+        var token = tree.Root.FindToken(absoluteIndex, includeWhitespace: true);
+
+        if (token.Kind is SyntaxKind.OpenAngle or SyntaxKind.CloseAngle)
```

Comment:
`<h1>$$Hello</h1>` looks like it returns the `CloseAngle`, so wouldn't show completions.

Reply:
Leaving open to remember to add a test

Reply:
Added some unit tests